### DISCUSSION TITLE
树莓派4B用VLESS-TCP+XTLS满载能到900M多一点，估计是上限了

### DISCUSSION THREAD

**Discussion by shenlijun** [XTLS/Xray-core#71] [Performance]
截图是实际线路测试结果，总体来说个人不推荐树莓派4B，如果其他可选的开发板能稳定运行的情况下，性价比都比树莓派要好.大家可以尝试下其它替代品

CPU没交授权费没有AES，预计发热问题也不能持久，改天试一下香橙派4B，那个据说有AES，U本身应该也强一些（它还在路上）
![捕获](https://user-images.githubusercontent.com/8860591/102090131-0a8d2b00-3e58-11eb-95d4-6da59fbfd988.PNG)

500m以下的线路应该没问题吧，但这种情况下树莓派性价比就不太行了。据说R2S之类的也能跑上来，当然俺都没试过-_-#
![捕获1](https://user-images.githubusercontent.com/8860591/102090984-2218e380-3e59-11eb-9877-94e99f45b74b.PNG)

**Comment by shenlijun** [comment]
splice的。

其实大家可以都把手头上的设备测试看看，这样后来者非土豪可以按需选择适合自己的

**Comment by badO1a5A90** [comment]
第一张图看起来榨干了。。。
话说这个成绩在一众硬件中相比较，应该也还算可以了吧

**Comment by shenlijun** [comment]
> 第一张图看起来榨干了。。。
> 话说这个成绩在一众硬件中相比较，应该也还算可以了吧

其实是可以的，比一些入门x86要强，但是总觉得有点高不成低不就，千兆差点意思，不跑千兆又有更便宜的替代

arm8标准要求要有aes，但是bcm2711没有，这个太骚了……

**Comment by bbcbbk** [comment]
树莓派4B现在只要200多了

**Comment by tritu** [comment]
Then this report should be run in chacha20 not AES. (When xtls can’t detect HasAES, it will use chacha20 cipher)

**Comment by shenlijun** [comment]
> 树莓派4B现在只要200多了

300多的香橙派更值得一试

**Comment by shenlijun** [comment]
> Then this report should be run in chacha20 not AES. (When xtls can’t detect HasAES, it will use chacha20 cipher)

Anyway， 900mb/s with 100% usage, maybe not a good choice yet

**Comment by RPRX** [comment]
https://www.youtube.com/watch?v=ePALlogM-SM

这位用树莓派 4B 跑 XTLS Direct 也到了 900Mb 左右，CPU 没满，是 60% 左右

而且还是开了 NAT 转发的 OpenWrt（他的 Splice 只比 Direct 强一点点），不知道是怎么做到的

**Reply by shenlijun** [reply]
> https://www.youtube.com/watch?v=ePALlogM-SM
> 
> 这位用树莓派 4B 跑 XTLS Direct 也到了 900Mb 左右，CPU 没满，是 60% 左右
> 
> 而且还是开了 NAT 转发的 OpenWrt（他的 Splice 只比 Direct 强一点点），不知道是怎么做到的

确实很奇怪，他那个视频里S905X3在xtls Direct 下面只有500m，CPU也没占满，按理说这个U应该不比bcm2711差多少，何况带宽更大的情况下无非也是U的占用更高，所以他这里可能是选的线路不一样了，这个不好说……

我测试用的是ubuntu server客户端，按理说比网关模式还要直接才对
另外一个香橙派4的rk3399一样armbian用客户端跑900+的时候cpu占用到了80%多，这个U各方面都比bcm2711强

难道是speedtest命令行会很占CPU？

**Reply by shenlijun** [reply]
今天两个开发板都没接在网络里，刚才用家里的网关测了下，跑到960m的情况下，htop里面显示xray占用60左右，speedtest占用15左右，按比例的话，就算去掉speedtest的影响也达不到他的效果。
或者是和我的证书有关系？
是Let's Encrypt申请的ec-256

**Reply by shenlijun** [reply]
或者是加密的时候选择了aes而不是chacha20？
但是rk3399可以硬解aes也占用到了80%

**Comment by shenlijun** [comment]
今天又用香橙派网关模式试了下，确实CPU占用会降下来，而且会一核繁忙，多核围观

本机直接speedtest会6个核心都到80%，速度能超过900m
网关模式会1个核心接近100%，其它5个40%左右，速度800多，而且居然还丢包

不知道是不是因为只用到一个核心，占用太高的时候丢了包
